src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/VulnerabilityAssessmentTests.ps1

```diff
@@ -0,0 +1,624 @@
+		Assert-AreEqual $scanRecord1FromListCmdlet.NumberOfFailedSecurityChecks $scanRecord1.NumberOfFailedSecurityChecks
+
+		# Run scan with piping
+		$excpectedScanCount = $excpectedScanCount + 1
+		Get-AzureRmSqlDatabase -ResourceGroupName $params.rgname -ServerName $params.serverName -DatabaseName $params.databaseName `
+		| Start-AzureRmSqlDatabaseVulnerabilityAssessmentScan -ScanId $scanId1
+
+		# Verify list scans with piping
+		$scanRecordList = Get-AzureRmSqlDatabase -ResourceGroupName $params.rgname -ServerName $params.serverName -DatabaseName $params.databaseName | Get-AzureRmSqlDatabaseVulnerabilityAssessmentScanRecord 
+		Assert-AreEqual $excpectedScanCount $scanRecordList.Count
+
+		$scanRecord1FromListCmdlet = $scanRecordList[$excpectedScanCount-1]
+		Assert-AreEqual $scanRecord1FromListCmdlet.ResourceGroupName $scanRecord1.ResourceGroupName
+		Assert-AreEqual $scanRecord1FromListCmdlet.ServerName $scanRecord1.ServerName
+		Assert-AreEqual $scanRecord1FromListCmdlet.DatabaseName $scanRecord1.DatabaseName
+		Assert-AreEqual $scanRecord1FromListCmdlet.ScanId $scanRecord1.ScanId
+		Assert-AreEqual $scanRecord1FromListCmdlet.TriggerType $scanRecord1.TriggerType
+		Assert-AreEqual $scanRecord1FromListCmdlet.State $scanRecord1.State
+
+
+		# Test WhatIf parameter of Start-AzureRmSqlDatabaseVulnerabilityAssessmentScan
+		$scanRecordList = Get-AzureRmSqlDatabaseVulnerabilityAssessmentScanRecord -ResourceGroupName $params.rgname -ServerName $params.serverName -DatabaseName $params.databaseName
+		$scansCount = $scanRecordList.Count
+
+		Start-AzureRmSqlDatabaseVulnerabilityAssessmentScan -ResourceGroupName $params.rgname -ServerName $params.serverName -DatabaseName $params.databaseName `
+		-WhatIf
+
+		# Verify no new scan was made
+		$scanRecordList = Get-AzureRmSqlDatabaseVulnerabilityAssessmentScanRecord -ResourceGroupName $params.rgname -ServerName $params.serverName -DatabaseName $params.databaseName
+		Assert-AreEqual $scansCount $scanRecordList.Count
+	}
+	finally
+	{
+		# Cleanup
+		Remove-VulnerabilityAssessmentTestEnvironment $testSuffix
+	}
+}
+
+<#
+.SYNOPSIS
+Tests for vulnerability assessment scan Convert scenarios 
+#>
+function Test-VulnerabilityAssessmentScanConvertTest
+{
+	# Setup
+	$testSuffix = getAssetName
+	Create-VulnerabilityAssessmentTestEnvironment $testSuffix
+	$params = Get-SqlVulnerabilityAssessmentTestEnvironmentParameters $testSuffix
+
+	try
+	{
+		# Set Vulnerability Assessment settings (SetSettingsWithStorageAccountNameParameterSet parameter set)
+		try
+		{
+			Set-AzureRmSqlDatabaseVulnerabilityAssessmentSettings -ResourceGroupName $params.rgname -ServerName $params.serverName -DatabaseName $params.databaseName `
+			 -StorageAccountName $params.storageAccount
+		}
+		catch
+		{
+			if ((Get-SqlTestMode) -eq 'Playback')
+			{
+				# This command fails in Playback mode because the recorder do not record storage container creation command
```

Comment:
This is not ideal behavior... Have you looked at why is the recorder not recording the storage container creation command?

Reply:
I couldn't find the reason that the recorder not recording the storage container creation command.
I changed the test to continue after the catch clause and by doing so the test is actually running and testing the all the cmdlet scenario.
In recording phase this command is passing so I'm not losing any test coverage.
As we get customers pressuring us to get those cmdlets, I'm keeping the code as is and opening a work item to investigate it later.